python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -536,3 +563,150 @@ def to_arrow(self):
                 arrow_polygons,
             ],
         )
+
+    def contains(self, other, align=True):
+        """Compute from a set of points and a set of polygons which points fall
+        within each polygon. Note that `polygons_(x,y)` must be specified as
+        closed polygons: the first and last coordinate of each polygon must be
+        the same.
+
+        This implements `.contains_properly`, which shares a large
+        space of correct cases with `GeoPandas.contains` but they do not
+        produce identical results. In the future we will use intersection
+        testing to match .contains behavior.
+
+        Parameters
+        ----------
+        test_points_x
+            x-coordinate of test points
+        test_points_y
+            y-coordinate of test points
+        poly_offsets
+            beginning index of the first ring in each polygon
+        poly_ring_offsets
+            beginning index of the first point in each ring
+        poly_points_x
+            x closed-coordinate of polygon points
+        poly_points_y
+            y closed-coordinate of polygon points
+
+        Examples
+        --------
+
+        Test if a polygon is inside another polygon:
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0.5, 0.5)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        dtype: bool
+
+
+        Test whether 3 points fall within either of two polygons
```

Comment:
There are only two different points and two different polygons in this example, but each one is repeated 3 times for a total of 6 tests. But there are only two unique tests in the example.

Reply:
You're right, I accidentally used two points and two polygons instead of what I said I was going to do.

Reply:
Now there are 6 different points and two different polygons, for a total of 6 different tests.

Reply:
Ugh, yes, thanks for your eyes on this example. Now it should have three unique points, each appearing twice, and two unique polygons appearing three times for pairwise comparison.